hack/ci-test.sh

```diff
@@ -20,8 +20,39 @@ set -o pipefail
 
 REPO_ROOT=$(dirname "${BASH_SOURCE}")/..
 
-cd $REPO_ROOT && \
-	source ./hack/fetch_ext_bins.sh && \
-	fetch_tools && \
-	setup_envs && \
-	make unit
+OPENSHIFT_CI=${OPENSHIFT_CI:-""}
+ARTIFACT_DIR=${ARTIFACT_DIR:-""}
+GINKGO=${GINKGO:-"go run ${REPO_ROOT}/vendor/github.com/onsi/ginkgo/v2/ginkgo"}
+GINKGO_ARGS=${GINKGO_ARGS:-"-v --randomize-all --randomize-suites --keep-going --race --trace --timeout=10m"}
+GINKGO_EXTRA_ARGS=${GINKGO_EXTRA_ARGS:-""}
+
+# Ensure that some home var is set and that it's not the root.
+# This is required for the kubebuilder cache.
+export HOME=${HOME:=/tmp/kubebuilder-testing}
+if [ $HOME == "/" ]; then
+  export HOME=/tmp/kubebuilder-testing
+fi
+
+if [ "$OPENSHIFT_CI" == "true" ] && [ -n "$ARTIFACT_DIR" ] && [ -d "$ARTIFACT_DIR" ]; then # detect ci environment there
+  GINKGO_ARGS="${GINKGO_ARGS} --junit-report=junit_cluster_machine_approver.xml --cover --coverprofile=test-unit-coverage.out --output-dir=${ARTIFACT_DIR}"
+fi
+
+# Print the command we are going to run as Make would.
+echo ${GINKGO} ${GINKGO_ARGS} ${GINKGO_EXTRA_ARGS} "<omitted>"
+${GINKGO} ${GINKGO_ARGS} ${GINKGO_EXTRA_ARGS} ./pkg/... ./cmd/...
```

Comment:
In this repo, we have tests right from the root, so better to just `./...` this one
```suggestion
echo ${GINKGO} ${GINKGO_ARGS} ${GINKGO_EXTRA_ARGS} ./...
${GINKGO} ${GINKGO_ARGS} ${GINKGO_EXTRA_ARGS} ./...
```